File: Models/PatientModel.cs
using System;

namespace PatientTestApplication.Models
{
    public class PatientModel
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Sex { get; set; }
        public DateTime Birthday { get; set; }
        public string DepartamentId { get; set; }
        public DPModel[] Diagnoses { get; set; }
        public DPModel[] Procedures { get; set; }
    }

    public class DPModel
    {
        public bool isActive { get; set; }
        public string Name { get; set; }
        public string[] Flags { get; set; }
        public string Description { get; set; }
    }
}

File: ClientApp/src/app/Models/PatientModel.ts
import {Sex} from "./Sex";

interface DPModel {
  isActive: boolean;
  name: string;
  flags: string[];
  description: string;
}

export interface PatientModel
{
  id: string;
  name: string;
  sex: Sex;
  birthday: Date;
  departamentId: string;
  diagnoses: DPModel[];
  procedures: DPModel[];
}
